src/IIIFingest/asset.py

```diff
@@ -116,13 +137,93 @@ def from_file(cls, filepath, **kwargs):
             metadata=metadata,
         )
 
+    @classmethod
+    def from_fileobj(cls, fileobj: BinaryIO, **kwargs) -> Asset:
+        """
+        Constructs an Asset from a file-like object.
+
+        Args:
+            fileobj:
+                A file-like object from which the asset will be generated.
+            **kwargs:
+                Several specific kwargs can be passed to the function, but if
+                they are omitted they will be inferred from the file. These
+                optional arguments are specified below.
+            asset_id:
+                A unique identifier for the asset. If none is specified, this
+                attribute of the Asset object output will be None.
+            width:
+                Width of the image in pixels. Ignored if height is not present.
+            height:
+                Height of the image in pixels. Ignored if width is not present.
+            format:
+                Mime type of the image. If not specified, this property can be
+                inferred from the content_type attribute on a Django
+                UploadedFile object, or directly from the file.
+            extension:
+                File extension for the image. If not specified, it is inferred
+                from the mime type.
+            label:
+                An optional label for the image. If none is specified, then the
+                asset_id will be used as a label.
+            metadata:
+                Metadata dictionary to be assigned to the asset.
+
+        Returns:
+            A newly constructed asset object.
```

Comment:
Nice docstring and type annotations!

Reply:
Thanks! I wanted to find a standard for docstrings that I mostly agree with, and I like [Google's approach](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#383-functions-and-methods), so I used it here.

Reply:
Yeah, I like Google's approach as well - it's a nice balance between the verboseness of NumPy docstrings and the terseness of ReST docstrings. I'd be in favor of adopting that in more of our projects.